flink-connector-sqlserver-cdc/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -0,0 +1,553 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.debezium.connector.sqlserver;
+
+import io.debezium.connector.sqlserver.SqlServerConnectorConfig.SnapshotMode;
+import io.debezium.pipeline.ErrorHandler;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.source.spi.StreamingChangeEventSource;
+import io.debezium.relational.Table;
+import io.debezium.relational.TableId;
+import io.debezium.schema.SchemaChangeEvent.SchemaChangeEventType;
+import io.debezium.util.Clock;
+import io.debezium.util.ElapsedTimeStrategy;
+import io.debezium.util.Metronome;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link StreamingChangeEventSource} based on SQL Server change data capture functionality. A
+ * main loop polls database DDL change and change data tables and turns them into change events.
+ *
+ * <p>The connector uses CDC functionality of SQL Server that is implemented as as a process that
+ * monitors source table and write changes from the table into the change table.
+ *
+ * <p>The main loop keeps a pointer to the LSN of changes that were already processed. It queries
+ * all change tables and get result set of changes. It always finds the smallest LSN across all
+ * tables and the change is converted into the event message and sent downstream. The process
+ * repeats until all result sets are empty. The LSN is marked and the procedure repeats.
+ *
+ * <p>The schema changes detection follows the procedure recommended by SQL Server CDC
+ * documentation. The database operator should create one more capture process (and table) when a
+ * table schema is updated. The code detects presence of two change tables for a single source
+ * table. It decides which table is the new one depending on LSNs stored in them. The loop streams
+ * changes from the older table till there are events in new table with the LSN larger than in the
+ * old one. Then the change table is switched and streaming is executed from the new one.
+ */
```

Comment:
This description about this class should contains the changes we do. See  [MySqlConnection.java line 40](https://github.com/ververica/flink-cdc-connectors/blob/master/flink-connector-mysql-cdc/src/main/java/io/debezium/connector/mysql/MySqlConnection.java#L40).